benchcab/workdir.py

```diff
@@ -28,60 +29,59 @@ def setup_src_dir(root_dir=internal.CWD):
-        )
-        os.makedirs(fluxsite_tasks_dir)
-
-    fluxsite_analysis_dir = Path(root_dir, internal.FLUXSITE_ANALYSIS_DIR)
-    if not fluxsite_analysis_dir.exists():
-        print(
-            f"Creating {fluxsite_analysis_dir.relative_to(root_dir)} directory: "
-            f"{fluxsite_analysis_dir}"
-        )
-        os.makedirs(fluxsite_analysis_dir)
-
-    fluxsite_bitwise_cmp_dir = Path(root_dir, internal.FLUXSITE_BITWISE_CMP_DIR)
-    if not fluxsite_bitwise_cmp_dir.exists():
-        print(
-            f"Creating {fluxsite_bitwise_cmp_dir.relative_to(root_dir)} directory: "
-            f"{fluxsite_bitwise_cmp_dir}"
-        )
-        os.makedirs(fluxsite_bitwise_cmp_dir)
-
-    print("Creating task directories...")
+    # Create the list of directories as sets because the order is not important
+    # and we want to avoid duplications.
+    fluxsite_paths = []
+
+    # Run directory
+    fluxsite_paths.append(Path(root_dir, internal.RUN_DIR))
+
+    # Fluxsite run directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_RUN_DIR))
+
+    # Fluxsite log directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_LOG_DIR))
+
+    # Fluxsite output directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_OUTPUT_DIR))
+
+    # Fluxsite tasks directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_TASKS_DIR))
+
+    # Fluxsite analysis directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_ANALYSIS_DIR))
+
+    # Fluxsite bit-wise comparison directory
+    fluxsite_paths.append(Path(root_dir, internal.FLUXSITE_BITWISE_CMP_DIR))
+
+    # Fluxsite tasks directories. append all of them as a set().
+    task_paths = []
     for task in fluxsite_tasks:
-        task_dir = Path(root_dir, internal.FLUXSITE_TASKS_DIR, task.get_task_name())
-        if not task_dir.exists():
-            if verbose:
-                print(f"Creating {task_dir.relative_to(root_dir)}: " f"{task_dir}")
-            os.makedirs(task_dir)
+        task_paths.append(
+            Path(root_dir, internal.FLUXSITE_TASKS_DIR, task.get_task_name()))
+    fluxsite_paths.append(task_paths)
+
+    return fluxsite_paths
+
+
+def setup_fluxsite_directory_tree(
+    fluxsite_paths: list[Union[Path, list]], root_dir=internal.CWD, verbose=False
+):
+    """Generate the directory structure used by `benchcab`."""
+
+    for work_path in fluxsite_paths:
+        if isinstance(work_path, list):
+            print("Creating task directories...")
+
+            for task_dir in work_path:
+                if verbose:
+                    print(f"Creating {task_dir.relative_to(root_dir)}: {task_dir}")
+                task_dir.mkdir(parents=True, exist_ok=True)
+        else:
+            print(
+                f"Creating {work_path.relative_to(root_dir)} directory: {work_path}"
+            )
+            work_path.mkdir(parents=True, exist_ok=True)
```

Comment:
We can put this code inside a utility function in [fs.py](https://github.com/CABLE-LSM/benchcab/blob/04dabaf556041fb905c1f3772f01ae8d34e01b79/benchcab/utils/fs.py) that wraps `Path.mkdir()`, i.e.
```suggestion
            mkdir(work_path.relative_to(root_dir), verbose=verbose, parents=True, exist_ok=True)
```
I think this would be useful in other places in the code base.